src/site/models/CoreModel.js

```diff
@@ -0,0 +1,124 @@
+const EventModel = require("./EventModel");
+const EventService = require("../services/EventService");
+
+class Entity
+{
+
+    constructor()
+    {
+        this.SUCCESS_EVENT_TYPE_CREATE_ENTITY = null;
+        this.SUCCESS_EVENT_TYPE_UPDATE_ENTITY = null;
+        this.SUCCESS_EVENT_TYPE_DELETE_ENTITY = null;
+    }
+
+    //implement it in children
+    getProvider(){
+        throw "Method should return schemaModel";
+    }
+
+    create(entityData)
+    {
+        //TODO:: validation
+        return new Promise( (resolve, reject) => {
+            this.getProvider().create(entityData, (err, newEntity) => {
+                if(err)
+                {
+                    //TODO:: events for error. If needed
+                    return reject(err);
+                }
+                this.__dispatchEventModel(this.SUCCESS_EVENT_TYPE_CREATE_ENTITY, newEntity);
+                resolve(newEntity);
+            });
+        });
+    }
+
+    update(id, entityData)
+    {
+        return new Promise( (resolve, reject) => {
+            this.getProvider().update({_id: id}, entityData, (err, updatedEntity) => {
+                if(err)
+                {
+                    return reject(err);
+                }
+                this.__dispatchEventModel(this.SUCCESS_EVENT_TYPE_UPDATE_ENTITY, updatedEntity);
+                resolve(updatedEntity);
+            });
+        });
+    }
+
+    remove(id)
+    {
+        return new Promise( (resolve, reject) => {
+            this.getProvider().remove({_id: id}, (err, removedData) => {
+                if(err)
+                {
+                    return reject(err);
+                }
+                this.__dispatchEventModel(this.SUCCESS_EVENT_TYPE_DELETE_ENTITY, {_id: id});
+                resolve(removedData);
+            });
+        });
+    }
+
+    getAll(paggingData = null, searchData = null)
+    {
+        return new Promise( (resolve, reject) => {
+            let query = this.getProvider().find(searchData, (err, collection) => {
+                if(err)
+                {
+                    return reject(err);
+                }
+                resolve(collection);
+            });
+            if(paggingData.limit)
+            {
+                query.limit(paggingData.limit);
+            }
+            if(paggingData.offset)
+            {
+                query.skip(paggingData.offset);
+            }
+        });
+    }
+
+    get(id)
+    {
+        return new Promise( (resolve, reject) => {
+            this.getProvider().findOne({_id: id}, (err, entity) => {
```

Comment:
`.findOne(...).exec()` - returns a promise already

Reply:
and ??? there should be some logic. error mapping for example

Reply:
it's already done by default,
check it please - http://mongoosejs.com/docs/promises.html#queries-are-not-promises

Reply:
get(id)
    {
        return new Promise( (resolve, reject) => {
            this.getProvider().findOne({_id: id}, (err, entity) => {
                if(err)
                {
                     /// create mapping error logic
                    return reject(err);
                }
                resolve(entity);
            });
        });
    }

Reply:
????

Reply:
@maximbondarenko90 
```
get(id) {
  return this.getProvider().findOne({_id: id}).exec()
}
```
in place of use:
```
const id = 'asd234fwfd'
try {
  const entity = await entity.get(id)
} catch (err) {
// do something with the error
}
```